server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -280,15 +280,16 @@ private void executeHealth(
         if (validationPredicate.test(currentState)) {
             ClusterHealthResponse clusterHealthResponse = getResponse(request, currentState, waitCount, TimeoutState.OK);
             if (request.ensureNodeWeighedIn() && clusterHealthResponse.hasDiscoveredClusterManager()) {
-                DiscoveryNode localNode = clusterService.state().getNodes().getLocalNode();
-                assert request.local() == true : "local node request false for request for local node weighed in";
-                boolean weighedAway = WeightedRoutingUtils.isWeighedAway(localNode.getId(), clusterService.state());
-                if (weighedAway) {
-                    listener.onFailure(new NodeWeighedAwayException("local node is weighed away"));
-                    return;
+                DiscoveryNode localNode = currentState.getNodes().getLocalNode();
+                if (localNode.isDataNode()) {
```

Comment:
Should it be made more generic based on roles?

Reply:
we can do that but can we say that all data nodes have `DATA_ROLE`? And ultrawarm nodes don't have `DATA_ROLE`

Reply:
added a `todo` to take this later